File: 14. Longest Common Prefix Easy/solution.java
import java.util.Arrays;

class solution {
    public static void main(String[] args) {
        System.out.println(longestCommonPrefix(new String[]{"flower","flow","flight"}));
        System.out.println(longestCommonPrefix(new String[]{"dog","racecar","car"}));
    }

    public static String longestCommonPrefix(String[] strs) {
        if (strs.length == 0) {
            return "";
        }

        Arrays.sort(strs);

        String first = strs[0];
        String last = strs[strs.length - 1];

        for (int i = 0; i < first.length(); i++) {
            if (first.charAt(i) != last.charAt(i)) {
                return first.substring(0, i);
            }
        }

        return first;
    }
}

File: 14. Longest Common Prefix Easy/solution.ts
function longestCommonPrefix(strs: string[]): string {
    if (!strs) return "";

    strs.sort();

    let first = strs[0];
    let last = strs[strs.length - 1];

    for(let i = 0; i < first.length; i++) {
        if (first[i] != last[i]) return first.substring(0, i);
    }

    return first;
}

console.log(longestCommonPrefix(["flower","flow","flight"]))
console.log(longestCommonPrefix(["dog","racecar","car"]))